pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -302,6 +326,8 @@ func (r *NSTemplateSetReconciler) ensureInnerNamespaceResources(logger logr.Logg
 	}
 	namespace.Labels[toolchainv1alpha1.RevisionLabelKey] = tcNamespace.Revision
 	namespace.Labels[toolchainv1alpha1.TierLabelKey] = nsTmplSet.Spec.TierName
+	namespace.Labels[toolchainv1alpha1.ProviderLabelKey] = toolchainv1alpha1.ProviderLabelValue
```

Comment:
You don't have to add this label here. It's already added in line 274: https://github.com/codeready-toolchain/member-operator/pull/142/files#diff-cffea82f7fdc7427b8c9f0d0fdd55aedR274

What you need to do instead is to add the label to all objects which are created inside the namespace. They are created in line 319: https://github.com/codeready-toolchain/member-operator/pull/142/files#diff-cffea82f7fdc7427b8c9f0d0fdd55aedL295
```
err = tmplProcessor.Apply(objs)
```

Reply:
@alexeykazakov i had another look at this code and the labels here actually belong to the objects prior to the apply. So in this case, the labels should be added just before the objects are created. `ensureNamespaceResource` actually ranges over the objects just the same way apply does so we can add labels to it. Maybe i misunderstood the code though?

Reply:
@alexeykazakov, i think you are right that they don't need to be added in both places though. will start up a temp cluster and test the idea to see if the labels still get set!

Reply:
@alexeykazakov i think it may be needed in both places - it looks like one func is called because namespace is nil and the other is called when namespace is not nil - is this correct? or did i miss something?

Reply:
@tinakurian Please, try to read the code one more time and check in which cases the `ensureNamespaceResource` or `ensureInnerNamespaceResources` are called. What do we check there - as you said, we check if the namespace is nil or not - but what does it refer to? 
Take a look also at the logic inside of these methods - what they do (their names should give you a hint). 
Check what kind of methods of the client is probably used in these methods (if `client.Update` method is called or rather `client.Create`)

If you focus on these parts of the code, then you should be able to clearly understand the purpose. This will help you with answering the question if the label has to be added here as well or not.

Reply:
@MatousJobanek @alexeykazakov fixed.. i understand... need to add a test for inner resources though..